radix-engine-common/src/data/scrypto/custom_serde.rs

```diff
@@ -0,0 +1,286 @@
+                    ),
+                },
+                Value::Custom {
+                    value: ScryptoCustomValue::InternalRef(InternalRef([0; OBJECT_ID_LENGTH])),
+                },
+            ],
+        };
+
+        let expected_simple = json!([
+            "resource_sim1qqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqz8qety",
+            {
+                "kind": "Own",
+                "value": "00000000000000000000000000000000000000000000000000000000000000"
+            },
+            "1",
+            "0.01",
+            "0",
+            {
+                "kind": "NonFungibleLocalId",
+                "value": "<hello>"
+            },
+            {
+                "kind": "NonFungibleLocalId",
+                "value": "#123#"
+            },
+            {
+                "kind": "NonFungibleLocalId",
+                "value": "[2345]"
+            },
+            {
+                "kind": "NonFungibleLocalId",
+                "value": "{1f52cb1e-86c4-47ae-9847-9cdb14662ebd}"
+            },
+            {
+                "kind": "Reference",
+                "value": "00000000000000000000000000000000000000000000000000000000000000"
+            }
+        ]);
+
+        let expected_invertible = json!({
+            "fields": [
+                {
+                    "kind": "Address",
+                    "value": "resource_sim1qqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqz8qety"
+                },
+                {
+                    "kind": "Own",
+                    "value": "00000000000000000000000000000000000000000000000000000000000000"
+                },
+                {
+                    "kind": "Decimal",
+                    "value": "1"
+                },
+                {
+                    "kind": "Decimal",
+                    "value": "0.01"
+                },
+                {
+                    "kind": "PreciseDecimal",
+                    "value": "0"
+                },
+                {
+                    "kind": "NonFungibleLocalId",
+                    "value": "<hello>"
+                },
+                {
+                    "kind": "NonFungibleLocalId",
+                    "value": "#123#"
+                },
+                {
+                    "kind": "NonFungibleLocalId",
+                    "value": "[2345]"
+                },
+                {
+                    "kind": "NonFungibleLocalId",
+                    "value": "{1f52cb1e-86c4-47ae-9847-9cdb14662ebd}"
+                },
+                {
+                    "kind": "Reference",
+                    "value": "00000000000000000000000000000000000000000000000000000000000000"
+                }
+            ],
+            "kind": "Tuple"
+        });
+
+        let context = ScryptoValueDisplayContext::with_optional_bench32(Some(&encoder));
+
+        assert_simple_json_matches(&value, context, expected_simple);
+        assert_invertible_json_matches(&value, context, expected_invertible);
```

Comment:
(Personal) I like `invertible_json` more than `simple_json`. :)

Reply:
Based on your thoughts I'll default the API to using the invertible format then 👍